website/content/docs/agent/config/config-files.mdx

```diff
@@ -914,6 +914,21 @@ Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'."
     - `agent_master` ((#acl_tokens_agent_master)) **Renamed in Consul 1.11 to
       [`acl.tokens.agent_recovery`](#acl_tokens_agent_recovery).**
 
+    - `config_file_service_registration` ((#acl_tokens_config_file_service_registration)) - The ACL token this agent
+      uses to register services and checks from [service definitions](/docs/discovery/services) and
+      [check definitions](/docs/discovery/checks) foudn in configuration files or in configuration
+      strings passed to the agent using the `-hcl` flag.
```

Comment:
Does registration also happen on reload?

If so, we could do something like:
```suggestion
      [check definitions](/docs/discovery/checks) loaded by the agent on startup and reload.
```

Reply:
nit: If we keep the `-hcl` flag description, would we also need to include `-json`?

Reply:
From what I can tell, `-json` doesn't exist. The `consul agent` command only has `-hcl` for config fragments.

Reply:
The `-hcl` flag enables operators to specify agent configuration values on the CLI. There is currently no equivalent `-json` flag for allowing agent configuration to be provided in JSON format. If we wanted to support that, it would be a new feature that requires additional development.

Reply:
I don't want to suggest we support that. For some reason I thought I had seen an invocation of a Consul agent with that recently, but it seems like I misremembered (e.g., perhaps it was a config file being passed in with JSON format).

Reply:
I think I prefer to be more elaborate / specific here to help reduce confusion?

If we say "configuration passed at startup", I feel like "startup" leaves room for interpretation. Is sending an HTTP request passing configuration to the agent? Does that include if I send a service registration request while the agent is "starting up"? 

I wanted to be clear about what services/checks the token is specifically used for (those services/checks in files or `-hcl` config fragments).